src/test/js/shared/idam_helper.js

```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
         });
     }
 
+    async  getToken() {
+        try {
+            // Check if TestData.TOKEN is not empty or null
+
+            if (TestData.FUNCTIONAL_TEST_TOKEN && TestData.FUNCTIONAL_TEST_TOKEN.trim() !== "") {
+                return TestData.FUNCTIONAL_TEST_TOKEN;
+            }
+            console.log("FUNCTIONAL_TEST_SERVICE_CLIENT_SECRET "+TestData.FUNCTIONAL_TEST_SERVICE_CLIENT_SECRET)
+
+            const response = await fetch(`${TestData.IDAM_API}/o/token`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                },
+                body: new URLSearchParams({
+                    grant_type: 'client_credentials',
+                    client_id: 'idam-functional-test-service',
+                    client_secret: TestData.FUNCTIONAL_TEST_SERVICE_CLIENT_SECRET,
+                    scope: 'profile roles',
+                }),
+            });
+
+            if (!response.ok) {
+                throw new Error('Failed to fetch token');
+            }
+
+            const tokenData = await response.json();
+            TestData.FUNCTIONAL_TEST_TOKEN = tokenData.access_token;
+            return tokenData.access_token; // Assuming token is in the 'access_token' field
+        } catch (error) {
+            console.trace('Error fetching token:', error);
+            throw error;
+        }
+    }
+
+    createServiceUsingTestingSupportService(serviceName, serviceClientSecret, roleId, token, scope = [], ssoProviders = '') {
+        let data;
+
+        if (roleId === '') {
```

Comment:
Don't like the amount of duplication here. Need to figure out what roleId is being used for in tests (see other comments - it's probably for private beta testing), and build the service object up as one set of code.